resource/identityschema/list_attribute.go

```diff
@@ -160,10 +161,13 @@ func (a ListAttribute) IsOptionalForImport() bool {
 // provider-defined implementation of the attribute to prevent unexpected
 // errors or panics. This logic runs during the GetResourceIdentitySchemas RPC and
 // should never include false positives.
-func (a ListAttribute) ValidateImplementation(ctx context.Context, req fwschema.ValidateImplementationRequest, resp *fwschema.ValidateImplementationResponse) {
+func (a ListAttribute) ValidateImplementation(_ context.Context, req fwschema.ValidateImplementationRequest, resp *fwschema.ValidateImplementationResponse) {
 	if a.CustomType == nil && a.ElementType == nil {
 		resp.Diagnostics.Append(fwschema.AttributeMissingElementTypeDiag(req.Path))
+		return
 	}
 
-	// TODO:ResourceIdentity: Write validation + tests that ensure the element type only contains primitive elements (bool, string, number)
+	if !fwtype.IsAllowedPrimitiveType(a.ElementType) {
```

Comment:
nit - We don't need to check this if it's a custom type
```suggestion
	if a.CustomType == nil && !fwtype.IsAllowedPrimitiveType(a.ElementType) {
```

Reply:
> We don't need to check this if it's a custom type

💭 Do we need another conditional here to validate a custom type against the allowed typables? Or is that somehow implicit?

Reply:
Well `a.ElementType` will be `nil` if it's a custom type which is why I mentioned that specifically here.

As for whether we should validate a custom type in general, we could although I wasn't as concerned about abusing a custom type to avoid those restrictions.

If we do want to validate the custom type, we would probably want to use the `.TerraformType()`, since that's ultimately what we send back to Terraform.

Reply:
Got it! Makes sense -- it took me another read to see that `a.CustomType` here is the concrete type of the `ListAttribute`, not the type of the list elements.